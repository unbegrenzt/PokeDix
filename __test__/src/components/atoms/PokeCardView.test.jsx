import React from 'react';
import { render, screen } from '@testing-library/react-native';
import { NativeBaseProvider } from 'native-base';

const { default: PokeCardView } = require('_components/atoms/PokeCardView');

const inset = {
  frame: {
    x: 0, y: 0, width: 0, height: 0,
  },
  insets: {
    top: 0, left: 0, right: 0, bottom: 0,
  },
};

describe('<PokeCardView />', () => {
  it('renders correctly', () => {
    const component = (
      <NativeBaseProvider initialWindowMetrics={inset}>
        <PokeCardView name="mew" />
      </NativeBaseProvider>
    );
    render(component);
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('jumps error fallback', () => {
    const component = (
      <PokeCardView name="mew" />
    );
    render(component);
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('error fallback return a valid message', async () => {
    const component = (
      <PokeCardView name="mew" />
    );
    render(component);

    const pokeNumber = await screen.findByText('151', { exact: false });
    const pokeHeader = await screen.findByText('Mew', { exact: false });
    const pokeType = await screen.findByText('psychic', { exact: false });

    expect(pokeNumber).toBeTruthy();
    expect(pokeHeader).toBeTruthy();
    expect(pokeType).toBeTruthy();
  });

  it('retrieve the pokemon info', async () => {
    const component = (
      <NativeBaseProvider initialWindowMetrics={inset}>
        <PokeCardView name="mew" />
      </NativeBaseProvider>
    );
    render(component);

    const pokeName = await screen.findByText('Mew', { exact: false });
    const pokeError = await screen.findByText('tap to retry', { exact: false });

    expect(pokeName).toBeTruthy();
    expect(pokeError).toBeTruthy();
  });
});
