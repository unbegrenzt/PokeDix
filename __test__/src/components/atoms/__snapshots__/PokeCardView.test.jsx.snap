// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PokeCardView /> jumps error fallback 1`] = `
<div
  role="alert"
>
  <p>
    Something went wrong:
  </p>
  <pre>
    \`useTheme\` must be used within \`NativeBaseConfigProvider\`
  </pre>
</div>
`;

exports[`<PokeCardView /> renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={Object {}}
    style={
      Object {
        "backgroundColor": "#0891b2",
        "borderRadius": 6,
        "maxWidth": "100%",
        "paddingBottom": 16,
        "paddingLeft": 12,
        "paddingRight": 12,
        "paddingTop": 16,
        "width": 375,
      }
    }
  >
    <View
      dataSet={Object {}}
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        dataSet={Object {}}
        style={
          Object {
            "justifyContent": "space-between",
          }
        }
      >
        <View
          dataSet={Object {}}
          style={Object {}}
        >
          <Text
            dataSet={Object {}}
            style={
              Object {
                "backgroundColor": undefined,
                "color": "#FFFFFF",
                "fontFamily": undefined,
                "fontSize": 14,
                "fontStyle": "normal",
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 21,
                "textDecorationLine": undefined,
              }
            }
          >
            Today @ 9PM
          </Text>
          <View
            dataSet={Object {}}
            style={
              Object {
                "height": 8,
              }
            }
          />
          <Text
            dataSet={Object {}}
            style={
              Object {
                "backgroundColor": undefined,
                "color": "#FFFFFF",
                "fontFamily": undefined,
                "fontSize": 20,
                "fontStyle": "normal",
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 30,
                "textDecorationLine": undefined,
              }
            }
          >
            Letâ€™s talk about avatar!
          </Text>
        </View>
        <View
          accessible={true}
          collapsable={false}
          dataSet={Object {}}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "flex-start",
              "backgroundColor": "#22d3ee",
              "borderRadius": 2,
              "paddingBottom": 4,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 4,
            }
          }
        >
          <Text
            dataSet={Object {}}
            style={
              Object {
                "backgroundColor": undefined,
                "color": "#FFFFFF",
                "fontFamily": undefined,
                "fontSize": 14,
                "fontStyle": "normal",
                "fontWeight": "700",
                "letterSpacing": 0,
                "lineHeight": 21,
                "textDecorationLine": undefined,
                "textTransform": "uppercase",
              }
            }
          >
            Remind me
          </Text>
        </View>
      </View>
      <View
        dataSet={Object {}}
        style={Object {}}
      />
      <Image
        accessibilityLabel="Aang flying and surrounded by clouds"
        alt="Aang flying and surrounded by clouds"
        dataSet={Object {}}
        onError={[Function]}
        source={
          Object {
            "uri": "https://media.vanityfair.com/photos/5ba12e6d42b9d16f4545aa19/3:2/w_1998,h_1332,c_limit/t-Avatar-The-Last-Airbender-Live-Action.jpg",
          }
        }
        style={
          Object {
            "borderRadius": 9999,
            "height": 100,
            "maxWidth": "100%",
            "width": 100,
          }
        }
      />
    </View>
  </View>
</RNCSafeAreaProvider>
`;
